homeassistant/components/hyperion/__init__.py

```diff
@@ -210,6 +322,18 @@ async def async_unload_entry(
         config_data = hass.data[DOMAIN].pop(config_entry.entry_id)
         for func in config_data[CONF_ON_UNLOAD]:
             func()
+
+        # Disconnect the shared instance clients.
+        await asyncio.gather(
+            *[
+                config_data[CONF_INSTANCE_CLIENTS][
+                    instance_num
+                ].async_client_disconnect()
+                for instance_num in list(config_data[CONF_INSTANCE_CLIENTS])
```

Comment:
No need to copy. This list comprehension is evaluated when it creates all the tasks before they are executed.

```suggestion
                for instance_num in config_data[CONF_INSTANCE_CLIENTS]
```